source/plugins/ruby/ApplicationInsightsUtility.rb

```diff
@@ -21,10 +21,15 @@ class ApplicationInsightsUtility
   @@EnvApplicationInsightsEndpoint = "APPLICATIONINSIGHTS_ENDPOINT"
   @@EnvControllerType = "CONTROLLER_TYPE"
   @@EnvContainerRuntime = "CONTAINER_RUNTIME"
-
+  @@isWindows = false
+  @@hostName = (OMS::Common.get_hostname)
```

Comment:
I think, we should avoid running this code - OMS::Common.get_hostname on windows since this linux specific implementation.

Reply:
We set in the yaml for windows - https://github.com/microsoft/Docker-Provider/blob/959b455d5ab873b6fa5ed7445bd61dc847ec2c08/kubernetes/omsagent.yaml#L779

Reply:
> I think, we should avoid running this code - OMS::Common.get_hostname on windows since this linux specific implementation.

This works in windows as well.

Reply:
How does the get_hostname method which has dependency "/var/opt/microsoft/docker-cimprov/state/containerhostname" path would work on windows? are you saying this returns just empty?  if its working then why need to read from environment variable?

Reply:
It returns the container host name, which is the pod name which is incorrect for our telemetry. It does not return empty.